packages/ripple/src/runtime/internal/client/runtime.js

```diff
@@ -278,6 +279,89 @@ export function derived(fn, block) {
   };
 }
 
+/**
+ * @param {any} v
+ * @param {TrackOptions | undefined} o
+ * @param {Block} b
+ * @returns {Tracked | Derived | Tracked[]}
+ */
+export function track(v, o, b) {
+  var is_tracked = is_tracked_object(v);
+
+  if (o === undefined) {
+    if (is_tracked) {
+      return v;
+    }
+
+    if (typeof v === 'function') {
+      return derived(v, b);
+    }
+    return tracked(v, b);
+  }
+
+  if (is_tracked || typeof v !== "object" || v === null || is_array(v)) {
+    throw new TypeError('Invalid value: expected a non-tracked object');
```

Comment:
Side note, we need to extract errors out at prod time, and use error codes instead. Otherwise our bundle size is going to blow up

Reply:
Yep, see how React and Svelte do it. I like how Svelte does it though, they go into a markdown file and get built to be imported.